lib/mediaType.js

```diff
@@ -120,15 +120,39 @@ function specify(type, spec, index) {
 
 }
 
-function preferredMediaTypes(accept, provided) {
+/**
+ * Returns a list of the acceptable media ranges in preferred order.
+ *
+ * @param {string} accept The raw contents of the `Accept` header
+ * @param {string[]} provided The media ranges the server can produce
+ *     for the resource in question.
+ * @param {Object} options Configuration options
+ * @param {boolean} options.detailed When `provided` is not defined, such
```

Comment:
> When `provided` is not defined

Is there a reason why we can't make this work for `provided` as well? To me, it seems like just returning these details objects for the matched types, which, if not directly useful, at least would make the API more straight-forward instead of needing to remember that sometimes the option does nothing (and yes, a lot of the current API bothers me, but eventually it can be rewritten ;) ).